test/extensions/filters/http/common/fuzz/uber_filter.h

```diff
@@ -60,6 +66,38 @@ class UberFilterFuzzer {
     ON_CALL(factory_context_.admin_, removeHandler(_)).WillByDefault(testing::Return(true));
   }
 
+  void addFileDescriptorsRecursively(const Protobuf::FileDescriptor* descriptor,
+                                     Protobuf::FileDescriptorSet* set,
+                                     absl::flat_hash_set<absl::string_view>* added_descriptors) {
+    if (!added_descriptors->insert(descriptor->name()).second) {
+      // Already added.
+      return;
+    }
+    for (int i = 0; i < descriptor->dependency_count(); i++) {
+      addFileDescriptorsRecursively(descriptor->dependency(i), set, added_descriptors);
+    }
+    descriptor->CopyTo(set->add_file());
+  }
+
+  void addProtoDescriptor(absl::string_view filter_name, Protobuf::Message* message) {
+    if (filter_name.find("grpc_json_transcoder") != absl::string_view::npos) {
+      envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder& config =
+          dynamic_cast<
+              envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder&>(
+              *message);
+      config.clear_services();
+      config.add_services("bookstore.Bookstore");
+
+      Protobuf::FileDescriptorSet descriptor_set;
+      const auto* file_descriptor =
+          Protobuf::DescriptorPool::generated_pool()->FindFileByName("test/proto/bookstore.proto");
+      ASSERT(file_descriptor != nullptr);
+      absl::flat_hash_set<absl::string_view> added_descriptors;
```

Comment:
The `added_descriptors`? It looks like I need to include all `bookstore.proto`s dependencies (any.proto, empty.proto, etc) recursively into the `FileDescriptorSet`, so I added a set to avoid repeating common deps.

Reply:
Yeah, it's just a bit confusing at the call site, since this is never used in the caller.

Reply:
Added a comment, i could also move this to the helper function, instantiating it if it's a nullptr as a starting point.